### PR TITLE
GEOMESA-1170 Re-implement KNN

### PR DESCRIPTION
**Pull request by elahrvivaz**
* Use an expanding query window instead of geohashes

Signed-off-by: Emilio Lahr-Vivaz <elahrvivaz@ccri.com>

### CODE REVIEW

File: geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/process/query/KNearestNeighborSearchProcessTest.scala [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+    ScalaSimpleFeature.create(sft, "pavilion V", "cville", "POINT( -78.504276 38.034971 )"),
+    ScalaSimpleFeature.create(sft, "pavilion VI", "cville", "POINT( -78.503424 38.034721 )"),
+    ScalaSimpleFeature.create(sft, "pavilion VII", "cville", "POINT( -78.504424 38.034628 )"),
+    ScalaSimpleFeature.create(sft, "pavilion VIII", "cville", "POINT( -78.503601 38.034343 )"),
+    ScalaSimpleFeature.create(sft, "pavilion IX", "cville", "POINT( -78.504617 38.034208 )"),
+    ScalaSimpleFeature.create(sft, "pavilion X", "cville", "POINT( -78.503833 38.033938 )"),
+    ScalaSimpleFeature.create(sft, "cabell", "cville", "POINT( -78.505152 38.032704 )"),
+    ScalaSimpleFeature.create(sft, "beams", "cville", "POINT( -78.510295 38.034283 )"),
+    ScalaSimpleFeature.create(sft, "mccormick", "cville", "POINT( -78.522288 38.032844 )"),
+    ScalaSimpleFeature.create(sft, "hep", "cville", "POINT( -78.520019 38.034511 )")
```
Comment (jnh5y) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/process/query/KNearestNeighborSearchProcessTest.scala]: Hehehehe.... did you know someone who worked in High Energy Physics?

File: geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geotools/package.scala [hunk 2]
```diff
@@ -28,6 +28,8 @@ package object geotools {
   val CRS_EPSG_4326: CoordinateReferenceSystem =
     try { CRS.decode("EPSG:4326", true) } catch { case t: Throwable => DefaultGeographicCRS.WGS84 }
 
+  val CrsEpsg4326: CoordinateReferenceSystem = CRS_EPSG_4326
```
Comment (jnh5y) [inline geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geotools/package.scala]: Anything wrong with the old val?

Comment (elahrvivaz) [reply geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geotools/package.scala]: nah, just trying to standardize on CamelCase constants.

File: geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/process/query/KNearestNeighborSearchProcessTest.scala (hunk 1 above)
Comment (ronq) [reply geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/process/query/KNearestNeighborSearchProcessTest.scala]: Hahaha you can tell I was still very excited about being back in town :)

Comment (ronq) [reply geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/process/query/KNearestNeighborSearchProcessTest.scala]: @jahhulbert-ccri take a gander at this IT test :)

**Pull request merged**